src/background.js

```diff
@@ -34,158 +36,394 @@
+  COOKIEBLOCK: "cookieblock",
+  DNT: "dnt",
+  USER_ALLOW: "user_allow",
+  USER_BLOCK: "user_block",
+  USER_COOKIE_BLOCK: "user_cookieblock",
+
+  // URLS
+  DNT_POLICIES_URL: "https://www.eff.org/files/dnt-policies.json",
+  COOKIE_BLOCK_LIST_URL: "https://www.eff.org/files/cookieblocklist_new.txt",
+
+  // The number of 1st parties a 3rd party can be seen on
+  TRACKING_THRESHOLD: 3,
+  MAX_COOKIE_ENTROPY: 12,
+
+  // Display debug messages
+  DEBUG: true,
+  INITIALIZED: false,
 
-// Instantiate privacy badgers grey list
-if (!("whitelistUrl" in localStorage)){
-  localStorage.whitelistUrl = "https://www.eff.org/files/cookieblocklist.txt";
-}
+  /**
+  * Per-tab data that gets cleaned up on tab closing
+    looks like:
+      tabData = {
+        <tab_id>: {
+          fpData: {
+            <script_origin>: {
+              canvas: {
+                fingerprinting: boolean,
+                write: boolean
+              }
+            },
+            ...
+          },
+          frames: {
+            <frame_id>: {
+              url: string,
+              parent: int
+            },
+            ...
+          },
+          trackers: {
+            domain.tld: bool
+            ...
+          }
+        },
+        ...
+      }
+  */
+  tabData: {},
 
-var whitelistUrl = localStorage.whitelistUrl;
-var isFirstRun = false;
-var seenDataCorruption = false;
-
-require("filterNotifier").FilterNotifier.addListener(function(action) {
-  // Called from lib/adblockplus.js after all filters have been created from subscriptions.
-  if (action == "load") {
-    // Update if newer version
-    var currentVersion = chrome.runtime.getManifest().version;
-    var prevVersion = localStorage.currentVersion;
-    if (prevVersion != currentVersion) {
-      migrateVersion(prevVersion, currentVersion);
-    }
-  }
-});
 
-// Load cookieblocklist and blocked domain listwhenever a window is created and whenever storage changes
-chrome.windows.onCreated.addListener(function(){
-  CookieBlockList.updateDomains();
-  BlockedDomainList.updateDomains();
+  // Methods
+  /**
+   * initialize privacy badger
+   */
+  init: function(){
+
+    pb.storage.initialize(function(){
+      if(pb.INITIALIZED) { return; }
+      pb.updateTabList();
+      pb.initializeDefaultSettings();
+      // TODO: Remove this when we remove ABP crap
+      require("filterNotifier").FilterNotifier.addListener(function(action) {
+        // Called from lib/adblockplus.js after all filters have been created from subscriptions.
+        if (action == "load") {
+          pb.runMigrations();
+          pb.initializeCookieBlockList();
+          pb.initializeDNT();
+        }
+      });
+
+      // Show icon as page action for all tabs that already exist
+      chrome.windows.getAll({populate: true}, function(windows) {
+        for (var i = 0; i < windows.length; i++) {
+          for (var j = 0; j < windows[i].tabs.length; j++) {
+            refreshIconAndContextMenu(windows[i].tabs[j]);
```

Comment:
let's say I have 100 tabs open and I install Privacy Badger, will this refresh all of my tabs?